index.js

```diff
@@ -626,14 +626,21 @@ class BitbucketScm extends Scm {
     * @return {Promise}
     */
     _getCheckoutCommand(config) {
-        const checkoutUrl = `https://${config.host}/${config.org}/${config.repo}`;
+        const checkoutUrl = `${config.host}/${config.org}/${config.repo}`;
+        const sshCheckoutUrl = `git@${config.host}:${config.org}/${config.repo}`;
         const checkoutRef = config.prRef ? config.branch : config.sha;
         const command = [];
 
         // Git clone
         command.push(`echo Cloning ${checkoutUrl}, on branch ${config.branch}`);
+        command.push('if [ ! -z $SCM_CLONE_TYPE] && [ $SCM_CLONE_TYPE = ssh ]; ' +
```

Comment:
Did some experiment, looks like `-n` and `! -z` behave differently:
When the var is not set, `-n $VAR` will still give u `true` 
```
 docker run -ti node:6 /bin/sh 
#  echo $SCM_CLONE_TYPE

# if [ ! -z $SCM_CLONE_TYPE ]; then echo "SCM_CLONE_TYPE len > 0"; fi
# if [ -n $SCM_CLONE_TYPE ]; then echo "SCM_CLONE_TYPE len > 0"; fi
SCM_CLONE_TYPE len > 0
# 
# if [ -n "$SCM_CLONE_TYPE" ]; then echo "SCM_CLONE_TYPE len > 0"; fi
# 
```
Adding double quote around the "$VAR" makes `-n` work fine. Not sure if we should go with that, since that will add more works compared with  just using `! -z`.